### DISCUSSION TITLE
need to do signature verification

### DISCUSSION THREAD

**Discussion by ryuk8000** [WalletConnect/walletconnect-monorepo#2393] [Developer Support] [answered]
async function WalletConnect(){
            await web3Modal.openModal()
            const signature = await window.wagmi.signMessage({
          message: 'gm wagmi frens',
          })
          
        // }
        //   console.log("signature",signature)
        }  
i am using web3Modal/html . How to await wallet connection before calling signMessage function.  
await web3modal.openModal() doesnt wait for wallet connection so signMessage gives error.

**Comment by xzilja** [comment]
You should use wagmi's action for this, I believe it was called `watchAccount`. Awaiting `openModal` just ensures that all data is ready before modal is displayed to the user.

**Reply by ryuk8000** [reply]
how to use watchAccount . Could you tell how to integrate with the above code that i have given

**Reply by xzilja** [reply, marked as answer]
You could start by reading about it here https://wagmi.sh/core/actions/watchAccount but for general programming questions, I'd suggest stackoverflow.

**Comment by adrianotadao** [comment]
Hey folks, I followed this example: https://docs.walletconnect.com/web3modal/react-native/hooks#usesignmessage but when I call signMessage() I was expecting to be redirected to the wallet app (eg. metamask) but this is not happening at all. Do I need something else?
